python/revng/daemon/__init__.py

```diff
@@ -34,7 +36,23 @@
 DEBUG = config("STARLETTE_DEBUG", cast=bool, default=False)
 
 
-def get_middlewares() -> List[Middleware]:
+class ManagerCredentialsMiddleware:
+    def __init__(self, app: ASGIApp, event_manager: EventManager):
+        self.app = app
+        self.event_manager = event_manager
+
+    async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
+        if scope["type"] != "http":
+            return await self.app(scope, receive, send)
```

Comment:
Can you make an example of a non-`http` request type that we expect?

Reply:
`lifetime` and `websocket`, per [ASGI Spec](https://asgi.readthedocs.io/en/latest/specs/main.html#protocol-specifications)

Starlette guarantees it's one of those 3 [here](https://github.com/encode/starlette/blob/master/starlette/routing.py#L729)

Reply:
I guess we'll have to do something about websockets one day, but not today.